R/htgts.R

```diff
@@ -0,0 +1,138 @@
+
+  # Type checking.
+  if (typeof(input_dir_path) != "character") {
+    stop(
+      paste(
+        "input_dir_path type is", paste0(typeof(input_dir_path), "."),
+        "It should be \"character\""
+      )
+    )
+  }
+  if (typeof(fastq1_name) != "character") {
+    stop(
+      paste(
+        "fastq1_name type is", paste0(typeof(fastq1_name), "."),
+        "It should be \"character\""
+      )
+    )
+  }
+  if (!is.null(fastq2_name) && typeof(fastq2_name) != "character") {
+    stop(
+      paste(
+        "fastq2_name type is", paste0(typeof(fastq2_name), "."),
+        "It should be \"character\""
+      )
+    )
+  }
+  if (typeof(expInfo_name) != "character") {
+    stop(
+      paste(
+        "expInfo_name type is", paste0(typeof(expInfo_name), "."),
+        "It should be \"character\""
+      )
+    )
+  }
+  if (!is.null(expInfo2_name) && typeof(expInfo2_name) != "character") {
+    stop(
+      paste(
+        "expInfo2_name type is", paste0(typeof(expInfo2_name), "."),
+        "It should be \"character\""
+      )
+    )
+  }
+  if (typeof(output_dir_name) != "character") {
+    stop(
+      paste(
+        "output_dir_name type is", paste0(typeof(output_dir_name), "."),
+        "It should be \"character\""
+      )
+    )
+  }
+  if (typeof(configType) != "character") {
+    stop(
+      paste(
+        "configType type is", paste0(typeof(configType), "."),
+        "It should be \"character\""
+      )
+    )
+  }
+  if (typeof(assembly) != "character") {
+    stop(
+      paste(
+        "assembly type is", paste0(typeof(assembly), "."),
+        "It should be \"character\""
+      )
+    )
+  }
+
+  # Check if input_dir_path exists
+  if (!is_running_in_docker()) {
+    if (!dir.exists(input_dir_path)) {
+      stop(paste("input_dir_path:", input_dir_path, "does not exist."))
+  }
+  if (!dir.exists(paste0(input_dir_path, "/", output_dir_name))) {
+      stop(paste("output_dir_path:", paste0(input_dir_path, "/", output_dir_name), "does not exist."))
+    }
+  }
+
+  # Executing the docker job
+  run_in_docker(
+    image_name = paste0("repbioinfo/htgts_pipeline_lts_v16:latest"),
+    volumes = list(
+      c(input_dir_path, "/Data")
+    ),
+    if (!is.null(fastq2_name)) {
+      additional_arguments = c(
+      "/Algorithm/HTGTS_Full.sh",
+        paste0("-fastq1 /Data/", fastq1_name),
+        paste0("-fastq2 /Data/", fastq2_name),
+        paste0("-expInfo /Data/", expInfo_name),
+        paste0("-expInfo2 /Data/", expInfo2_name),
+        paste0("-outDir /Data/", output_dir_name),
+        paste("-configType", configType),
+        paste ("-assembly", assembly)
+        )
+      }
+    else {
+      additional_arguments = c(
+      "/Algorithm/HTGTS_Full.sh",
+        paste0("-fastq1 /Data/", fastq1_name),
+        paste0("-expInfo /Data/", expInfo_name),
+        paste0("-outDir /Data/", output_dir_name),
+        paste("-configType", configType),
+        paste ("-assembly", assembly)
+        )
+      }
```

Comment:
I'm not extremely sure about this form. I would take the additional arguments out of the `run_in_docker` function call.